src/components/TextInputComponent/TextInputComponent.tsx

```diff
@@ -0,0 +1,72 @@
+import { forwardRef, type InputHTMLAttributes, type KeyboardEventHandler, useMemo } from 'react';
```

Comment:
forwardRef is deprecated. Please "ref"

https://react.dev/reference/react/forwardRef